src/flownet/network_model/_network_model.py

```diff
@@ -50,6 +50,8 @@ def __init__(
         if isinstance(fault_planes, pd.DataFrame):
             self._fault_planes: pd.DataFrame = fault_planes
```

Comment:
As this can be optional, we will then have to change the type annotation:

```suggestion
            self._fault_planes: Optional[pd.DataFrame] = fault_planes
```